pkg/database/audit_entry.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"time"
+)
+
+// AuditEntry represents an event in the system. These records are purged after
+// a configurable number of days by the cleanup job. The AuditEntry specifically
+// does NOT make use of foreign keys or relationships to avoid breaking an audit
+// entry if the upstream data which was audited is removed or changed. These
+// records should be considered immutable.
+type AuditEntry struct {
+	Errorable
+
+	// ID is the entry's ID.
```

Comment:
any reason why not the gorm.Model embed?

Reply:
`gorm.Model` changes the behavior and semantics for delete. If the resource has `DeletedAt`, it soft deletes. Audit entries are also immutable and `gorm.Model` includes an unnecessary `UpdatedAt` field. Given I expect audit_entries to be our largest table (even with purging), I wanted to keep the number of columns as small as possible.